engine/runtime/src/main/java/org/enso/interpreter/runtime/EnsoPolyglotJava.java

```diff
@@ -180,30 +189,46 @@ static void addToClassPath(EnsoContext ctx, Object whoIsIgnored, File path)
     data.guest.addToClassPath(path);
   }
 
+  private void acquireLock() {
+    while (!lock.tryAcquire()) {
+      TruffleSafepoint.poll(null);
```

Comment:
- Is this `TruffleSafepoint.poll` essential part of the fix?
- I guess it is...
- ...just `lock.acquire` wouldn't fix the problem I assume